src/controllers/products.ts

```diff
@@ -0,0 +1,82 @@
+import Product from '../models/products';
+import {
+  createProductParams,
+  CreateProductParams,
+  GetProductsParams,
+  UpdateProductParams,
+} from '../interfaces/products';
+import { undefined } from 'zod';
+import { Op } from 'sequelize';
+
+export const getProducts = async ({ name, categories, sortBy, page, size }: GetProductsParams) => {
+  const totalCount = await Product.count();
+  type obj = {
+    [key: string]: any;
+  };
+
+  const where: obj = {};
+
+  if (name != null) {
+    where.name = { [Op.like]: `%${name}%` };
+  }
+
+  if (categories != null) {
+    where.categoryId = { [Op.in]: [categories] };
+  }
+
+  const products1 = await Product.findAndCountAll();
+
+  const products = await Product.findAndCountAll({
+    where,
+    order:
+      sortBy === 'price_asc' ? [['price', 'ASC']] : sortBy === 'price_desc' ? [['price', 'DESC']] : [['id', 'ASC']],
+    offset: page * size,
+    limit: size,
+    subQuery: false,
+  });
```

Comment:
I think the double ternary is not really clear, maybe you can do something like

```suggestion
 const sortByToOrder = [
   price_asc: [['price', 'ASC']],
   price_desc: [['price', 'DESC']]
 ]
 
  const products = await Product.findAll({
    where,
    order: sortByToOrder[sortBy] ?? [['id', 'ASC']],
    offset: page * size,
    limit: size,
  });
```